### DISCUSSION TITLE
how to change USB MIDI Device's name

### DISCUSSION THREAD

**Discussion by meduson** [tttapa/Control-Surface#736] [General]
Hi there

I'd like to change the midi Device's name of my arduino home made control surfaces. I've seen a post with a trick to do it for a BLE MIDI device :

> I've just pushed an update, you can now add MIDI_BLE.setName("My Midi Device"); to your setup (before Control_Surface.begin()).

is there a way to do the same with USB devices ? I've several and it would be easier to identify them.

thanks in advance for your help

**Comment by tttapa** [comment]
For Teensy, see:
- https://github.com/tttapa/Control-Surface/issues/555#issuecomment-903181997
- https://github.com/tttapa/Control-Surface/issues/254#issuecomment-671478464

AFAIK, you cannot dynamically change the name of the USB interfaces without resetting and going through enumeration again.

If you're not using a Teensy, see the documentation and issues for the [MIDIUSB](https://github.com/arduino-libraries/MIDIUSB) library.

**Reply by meduson** [reply]
Thanks for your answer. Unfortunately, I'm nos using Teensy but arduino micro. Is it possible to do this with a regular arduino ?

**Reply by tttapa** [reply]
See https://github.com/arduino-libraries/MIDIUSB/issues?q=is%3Aissue+name

**Comment by meduson** [comment]
thank you very much for the link !
